.github/scripts/close-invalid-link.cjs

```diff
@@ -23,21 +23,25 @@ async function closeIssue() {
 module.exports = async ({ github, context }) => {
     const owner = context.repo.owner;
     const repo = context.repo.repo;
-    const number = context.issue_number;
+    const number = context.issue.number;
 
     const issue = await github.rest.issues.get({
       owner: owner,
       repo: repo,
       issue_number: number,
     });
 
-    try {
-        const link = issue.data.body.match(/(https?:\/\/github.com\/.*)/)[0];
-        const isValidLink = (await fetch(link)).ok;
-        if (!isValidLink) {
-           await closeIssue();
+    const isBugTemplate = issue.data.body.includes("Link to the code that reproduces this issue");
+
+    if (isBugTemplate) {
+        try {
+            const link = issue.data.body.match(/(https?:\/\/github.com\/.*)/)[0];
```

Comment:
A bit out of scope for this PR, but would this capture references with `#` (e.g. issue `#12321`)?

Reply:
What if there are multiple links? Should they all be valid, at least one, or just the first?

If not just the first, perhaps `String#matchAll` would be nice:
```ts
let foundValid = false;

for (const link of issue.data.body.matchAll(/https?:\/\/\w*/g)) {
  try {
    const url = new URL(link);
    const validHostname = url.hostname.endsWith('github.com');

    if (!validHostname) continue;
    if ((await fetch(url)).ok) {
      foundValid = true;
      break;
    }
  } catch {
    continue;
  }
}
```

Reply:
I think just one should be fine! Perhaps people want to include other links that are somewhat relevant but not faithful reproductions of the issue

Reply:
Also, as to your first point: yes, technically it would match `issues`, `pulls`, etc. However, I'm thinking there's one of two cases: either 1) they accidentally linked to pulls, etc., of a real reproduction repo, in which case we can just navigate to it, or 2) they linked to a non-reproduction repo's pulls, issues, etc. If 1, we should still review. If 2, then there's nothing stopping them from just submitting a github repo that isn't actually a reproduction repo (i.e., submitting something like google-cloud-node). Hopefully this is the minority of cases, and we can just close those manually.

Reply:
Nice, sg  👌🏽